### PR TITLE
Implement `Bake` for VZV types of different formats

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: utils/zerovec/src/lib.rs [hunk 1]
```diff
@@ -291,6 +291,11 @@ pub mod vecs {
     pub use crate::varzerovec::{VarZeroSlice, VarZeroVec};
 
     pub use crate::varzerovec::{Index16, Index32, Index8, VarZeroVecFormat, VarZeroVecOwned};
+
+    pub type VarZeroVec16<'a, T> = VarZeroVec<'a, T, Index16>;
```
Comment (Manishearth) [inline utils/zerovec/src/lib.rs:295]: nit: not sure I like these reexports.

File: utils/zerovec/src/varzerovec/databake.rs [hunk 2]
```diff
@@ -2,42 +2,85 @@
...
     fn bake(&self, env: &CrateEnv) -> TokenStream {
         env.insert("zerovec");
         if self.is_empty() {
-            quote! { zerovec::VarZeroVec::new() }
+            quote! { zerovec::vecs::VarZeroVec16::new() }
         } else {
             let bytes = databake::Bake::bake(&self.as_bytes(), env);
             // Safe because self.as_bytes is a safe input
-            quote! { unsafe { zerovec::VarZeroVec::from_bytes_unchecked(#bytes) } }
+            quote! { unsafe { zerovec::vecs::VarZeroVec16::from_bytes_unchecked(#bytes) } }
```
Comment (Manishearth) [inline utils/zerovec/src/varzerovec/databake.rs:17]: nit: we should just use VZV here, the generic gets defaulted

Comment (Manishearth) [reply utils/zerovec/src/varzerovec/databake.rs:17]: thought: we could implement for all formats and then do a typeid type check to emit the compact thing for VZV16

Comment (robertbastian) [reply utils/zerovec/src/varzerovec/databake.rs:17]: It doesn't, that's the problem. It gets inferred, so it's very unsafe to not specify it.

Comment (Manishearth) [reply utils/zerovec/src/varzerovec/databake.rs:17]: @robertbastian hmm, I guess.

**Review by Manishearth — APPROVED** [review]
sold on the safety angle

**Review by sffc — COMMENTED** [review]
LGTM. I've experienced hard-to-debug safety issues involving VZVs that change between index types. I like the re-exports because it keeps the AST the same size.

**Pull request merged**